src/tools/pkcs11-tool.c

```diff
@@ -3946,6 +3946,45 @@ show_key(CK_SESSION_HANDLE sess, CK_OBJECT_HANDLE obj)
 			}
 		}
 		break;
+	case CKK_EC_EDWARDS:
+		printf("; EC_EDWARDS");
+		/* fall through */
+	case CKK_EC_MONTGOMERY:
+		if (key_type == CKK_EC_MONTGOMERY) {
+			printf("; EC_MONTGOMERY");
+		}
```

Comment:
Though this is a matter of taste, wouldn't it be more readable if you write without fallthrough:
```c
case CKK_EC_EDWARDS:
case CKK_EC_MONTGOMERY:
if (key_type == CKK_EC_EDWARDS) {
  printf("; EC_EDWARDS");
} else {
  printf("; EC_MONTGOMERY");
}
```